Disher/View Controllers/RecipesViewController.m

```diff
@@ -224,18 +236,56 @@ - (void) queryAPIs:(NSString *) input withOption:(NSInteger) option completionHa
     dispatch_group_async(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
         [self querySearchSpoonacular:input withOption:option completionHandler:^(NSArray *returnedMeals) {
             self.spoonResults = returnedMeals;
+            for (NSDictionary *meal in returnedMeals) {
+                NSString *recipeName;
+                NSString *imageLink;
+                NSString *mealID;
+                NSString *source;
+                NSArray *cuisine;
+                recipeName = meal[@"title"];
+                imageLink = meal[@"image"];
+                mealID = [NSString stringWithFormat:@"%@", meal[@"id"]];
+                source = @"Spoonacular";
+                if (((NSArray *)meal[@"cuisines"]).count) {
+                    cuisine = meal[@"cuisines"];
+                    for (NSString *cuisine in meal[@"cuisines"]) {
+                        [self.cuisines addObject:cuisine];
+                    }
+                }
+                else {
+                    cuisine = @[@"Unknown"];
+                }
+                Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisine];
+                [self.tableViewRecipes addObject:newRecipe];
+            }
             dispatch_group_leave(group);
         }];
     });
     dispatch_group_enter(group);
     dispatch_group_async(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
         [self queryMealDB:input withOption:option completionHandler:^(NSArray *returnedMeals) {
             self.mealDBresults = returnedMeals;
+            NSString *recipeName;
+            NSString *imageLink;
+            NSString *mealID;
+            NSString *source;
+            NSArray *cuisine;
+            for (NSDictionary *meal in returnedMeals) {
+                recipeName = meal[@"strMeal"];
+                imageLink = meal[@"strMealThumb"];
+                mealID = meal[@"idMeal"];
+                source = @"TheMealDB";
+                cuisine = @[meal[@"strArea"]];
+                [self.cuisines addObject:meal[@"strArea"]];
```

Comment:
should we use the `cuisine` variable instead of `meal[@"strArea"]` again?

Reply:
I did this because I am storing an array to unify it with spoonacular results. MealDB returns the cuisine as a string and spoonacular returns an array so a compromise I made was to make both into arrays